### DISCUSSION TITLE
The ReprocessScheduler is not working for stations with an IP address

### DISCUSSION THREAD

**Discussion by SamanthaKarpyshin** [opendcs/opendcs#137] [Q&A]
Hello I am new to OPENDCS so forgive my limited knowledge. My problem is with the ReprocessScheduler for Reprocess-EDL. 
I can connect to meteorological stations and have raw data files. They decode well with the current config I have but I am not able to reprocess the data and decode it into our database. I cannot even bring in raw data into the DCP message browser either to ensure it is fully decoding. For example say I have a data gap that I am trying to fill from Nov 15-Dec 3. I will be able to connect to the station and get all the raw data from Nov 15-Dec 3. When I try and use the ReprocessScheduler it may only reprocess and decode from Dec 1-3. Any other date than those ~48 hours and it will not reprocess it. The raw data has not changed its format. 

I have a few screen shots:
 
Raw files received spanning Nov 15-Dec3:

![rawmessages](https://user-images.githubusercontent.com/120065688/206318899-422c9b1e-cb9c-41d1-80ff-cefd3dad65bc.png)

Settings on Reprocess-edl stations:

![reprocessscheduler2](https://user-images.githubusercontent.com/120065688/206319263-2a05ea52-5c5d-45b1-9d5b-eee03e7bd1c2.png)

No raw or decoded message in DCP message browser:

![dcpmessage](https://user-images.githubusercontent.com/120065688/206319752-60e9df24-4c3b-40be-a4b9-1cb40e89b40b.png)

Here is the error message:
![error message](https://user-images.githubusercontent.com/120065688/206320855-9c48ad5a-d9d3-45fe-9a7a-69449d92bb6e.jpg)

Thanks for any help with this,
Sam

**Comment by agilmore2** [comment]
That routing spec looks like it's reading from an LRGS and writing to a zrxp file. What's reading the EDL data and writing to LRGS?

That last error message looks like a serious LRGS error. Are there other issues with the Dec 6 data?

**Comment by mikecovesw** [comment]
I agree with Andrew. That error indicates that the Dec 6 archive file is corrupt at the byte position indicated. The data after that position may not be recoverable, or at least not easily. 

You say you're trying to get data from Nov 15 through Dec 3. So why is the routing spec time range set to Dec 6 & 7?
Also, you should uncheck all the boxes under "Platform/Message Types". You don't need them because you're specifying a particular platform by its ID. Also, I don't know what it will do when you say you want an EDL message from GOES East. In general, unchecking everything means the filter is wide open and you'll get everything.

**Comment by SamanthaKarpyshin** [comment]
Thank you for the information. I will try those things. As far as Dec 6 and
7th goes I was just experimenting with dates I already have decoded and
ingested to see if the ReprocessingScheduler would reprocess those. No
luck. Dec 6th has data and is decoded and ingested so I am not sure what
that error is suggesting for that particular date.

Thanks
Sam

**Comment by mikecovesw** [comment]
Do you get the same error for Nov 15 through Dec 3? Or is it a different 
problem?

On 12/8/22 8:25 AM, SamanthaKarpyshin wrote:
> Thank you for the information. I will try those things. As far as Dec 
> 6 and
> 7th goes I was just experimenting with dates I already have decoded and
> ingested to see if the ReprocessingScheduler would reprocess those. No
> luck. Dec 6th has data and is decoded and ingested so I am not sure what
> that error is suggesting for that particular date.
>
> Thanks
> Sam
>
> On Thu., Dec. 8, 2022, 6:05 a.m. Mike Maloney, ***@***.***>
> wrote:
>
> > I agree with Andrew. That error indicates that the Dec 6 archive file is
> > corrupt at the byte position indicated. The data after that position may
> > not be recoverable, or at least not easily.
> >
> > You say you're trying to get data from Nov 15 through Dec 3. So why 
> is the
> > routing spec time range set to Dec 6 & 7?
> > Also, you should uncheck all the boxes under "Platform/Message 
> Types". You
> > don't need them because you're specifying a particular platform by 
> its ID.
> > Also, I don't know what it will do when you say you want an EDL message
> > from GOES East. In general, unchecking everything means the filter 
> is wide
> > open and you'll get everything.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/opendcs/opendcs/discussions/137#discussioncomment-4343088>,
> > or unsubscribe
> > 
> <https://github.com/notifications/unsubscribe-auth/A4UA5GBNWXGB2YJDH6MVCZTWMHMIZANCNFSM6AAAAAASXOA6OY>
> > .
> > You are receiving this because you authored the thread.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/opendcs/opendcs/discussions/137#discussioncomment-4343233>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ABTDKHT7WLHPXWHPMPW75CTWMHOVBANCNFSM6AAAAAASXOA6OY>.
> You are receiving this because you commented.Message ID: 
> ***@***.***>
>
-- 
Michael Maloney
Cove Software, LLC
Tel: (410) 241-5248
***@***.***
Web:www.covesw.com

**Reply by SamanthaKarpyshin** [reply]
Thanks for the replies. When I reprocess without any platform/message type selected (pic1), I receive the following messages (pic2), and I only get reprocessed data up until Nov 22 (pic3). 
![pic1](https://user-images.githubusercontent.com/120065688/206496797-6fb3d1c9-a636-4ea9-951b-1259cd6d19dd.png)
![pic2](https://user-images.githubusercontent.com/120065688/206496831-b2fd1786-526a-4869-9d70-c5cd6a5dafdd.png)
![pic3](https://user-images.githubusercontent.com/120065688/206496853-bb7213e2-ff4c-465d-aa60-9a6296d1ffde.png)

Also when I am polling data to fill the gap, I also get the message is too big. Would this causes the messages in the LRGS to be corrupted?
![warning from polling](https://user-images.githubusercontent.com/120065688/206498141-880e5f01-6637-4fb9-b0b6-2040de1d7b8c.jpg)

Thanks again
Sam

**Comment by mikecovesw** [comment]
The WARNING message indicates that, yes, the PAKBUS polling session resulted in a message that was bigger than the maximum. The limit is due to the DDS protocol which uses 5 digits to represent message length plus header.
The the message indicates that the software truncated the file, so all should be well.
One question for you to explore is why is that Campbell station returning such a large amount of data.

As for the issue of not getting data past nov 22, I would experiment with the time ranges and see if you can run the routing spec day-by-day.
